octopus/src/test/scala/octopus/example/domain/BaseAsyncMonadSpec.scala

```diff
@@ -0,0 +1,87 @@
+package octopus.example.domain
+
+import octopus.dsl._
+import octopus.syntax._
+import octopus.{AsyncValidator => _, _}
+import org.scalatest.{AsyncWordSpec, MustMatchers}
+
+import scala.concurrent.Future
+
+trait BaseAsyncMonadSpec[M[_]] extends Fixtures with MustMatchers { this: AsyncWordSpec =>
```

Comment:
I suggest organizing this test suite better, eliminating few files at the same time. I don't know what is async monad, but it is part of the suite name. My feeling is that you trying to organize tests that are parameterized by `M[_]`, but they cover somehow different part than `AsyncValidationSpec`. 

I suggest to merge these two suites into a single one, called `abstract class AsyncValidationSpec[M[_]: App] extends WordSpec with ...`, which contains all the tests written using word-spec style, organized in tree looking from the top-level - this would enable you to get rid of def`s like `validateResultWithAsyncValidator` or `validateSimpleEmail` which you had to invoke by hand.

This abstract class is not a test suite yet, because it requires parameterization, which can be done even in the same file:

```
case Future_AsyncValidationSpec extends AsyncValidationSpec[Future]
case IO_AsyncValidationSpec extends AsyncValidationSpec[IO] // this probably in scalaz-effect module
// etc.
```

This will make these tests more cleaner to read and instantiate.